src/MarkdownTextInput.tsx

```diff
@@ -36,15 +36,35 @@ function processMarkdownStyle(input: PartialMarkdownStyle | undefined): Markdown
   return processColorsInMarkdownStyle(mergeMarkdownStyleWithDefault(input));
 }
 
-const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>((props, ref) => {
+const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(({value, maxLength, onChangeText, ...props}, ref) => {
   const IS_FABRIC = 'nativeFabricUIManager' in global;
 
   const markdownStyle = React.useMemo(() => processMarkdownStyle(props.markdownStyle), [props.markdownStyle]);
 
+  const onChangeTextRef = React.useRef(onChangeText);
+  onChangeTextRef.current = onChangeText;
+
+  const processedText = React.useMemo(() => {
+    if (maxLength !== undefined) {
+      return value?.slice(0, maxLength);
+    }
+    return value;
+  }, [value, maxLength]);
+
+  React.useLayoutEffect(() => {
+    if (maxLength === undefined || !value || value.length <= maxLength || !onChangeTextRef.current) {
+      return;
+    }
+    onChangeTextRef.current(value.slice(0, maxLength));
+  }, [value, maxLength]);
+
   return (
     <>
       <TextInput
         {...props}
+        value={processedText}
+        onChangeText={onChangeText}
```

Comment:
What if the component was supposed to be uncontrolled? I don't really like this logic here